src/runner/test-run-controller.js

```diff
@@ -43,40 +55,41 @@ export default class TestRunController extends EventEmitter {
     }
 
     _createTestRun (connection) {
-        var quarantineAttemptNum = this.quarantine ? this.quarantine.attemptNumber : null;
-        var screenshotCapturer   = this.screenshots.createCapturerFor(this.test, this.index, quarantineAttemptNum, connection, this.warningLog);
-        var TestRunCtor          = this.TestRunCtor;
+        var screenshotCapturer = this.screenshots.createCapturerFor(this.test, this.index, this.quarantine, connection, this.warningLog);
+        var TestRunCtor        = this.TestRunCtor;
 
         this.testRun = new TestRunCtor(this.test, connection, screenshotCapturer, this.warningLog, this.opts);
 
+        if (this.testRun.addQuarantineInfo)
+            this.testRun.addQuarantineInfo(this.quarantine);
+
         return this.testRun;
     }
 
     async _endQuarantine () {
-        if (this.quarantine.attemptNumber > 1)
-            this.testRun.unstable = this.quarantine.passedTimes > 0;
+        if (this.quarantine.attempts > 1)
```

Comment:
`this.quarantine.attempts.length > 1` ? Do we have tests for unstable/stable tests?)

Reply:
thanks, looks like we don't